### PR TITLE
fix: Radio.Button title loss

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/43008

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix Radio.Button title loss，update typescript  definition    |
| 🇨🇳 Chinese |     修复单选按钮 title 属性不生效，补齐 ts     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 418a4a2</samp>

Added a `title` prop to the `Checkbox` and `Radio` components and their documentation files. This prop allows setting a custom tooltip for the checkbox or radio button, using the native HTML `title` attribute. Updated the documentation in both English and Chinese with version and title information for the new prop.

### <samp>🤖 Generated by Copilot at 418a4a2</samp>

*  Add `title` prop to `Checkbox` and `Radio` components and their documentation ([link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86R19), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86R65), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86R139), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-d3e984f0984d6a44c780ab883b2f78b2df8029b1ce515d879e60879bb425fa63R16), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-5b7813283fde6431f10030c6ace1984dc18e77251c8809423c1b41232a92466fL35-R35), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-5b7813283fde6431f10030c6ace1984dc18e77251c8809423c1b41232a92466fR41), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-e387f360a3bed763861e56375e36712dacf05631b3466b2236a084bf4d074a9cL36-R36), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-e387f360a3bed763861e56375e36712dacf05631b3466b2236a084bf4d074a9cR42), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-94794a2156476ac265df2274830a3c700b50b66ef4bf24f7558ab3c2885f7749R80), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-284376d0613bd5179c694e5ba3bd52b7272adba2b80f03c1bf281c57ada16f78L40-R40), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-284376d0613bd5179c694e5ba3bd52b7272adba2b80f03c1bf281c57ada16f78R46), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-a67ba6e2ccf07e742a81b5b32138cf7510a2b5eff68a6498e79aa0487c09bb31R48), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-4befd0a047da31184c43bc6fef846467a6e774ecef3cf332ac25310b3d9d4a8cR81))
* Adjust import order of modules in `Checkbox.tsx` and `radio.tsx` ([link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-e5138148a71ae8aea8e59e4b8934815b003ff8836bbb6ca6dc91d41aef32ef86L5-R9), [link](https://github.com/ant-design/ant-design/pull/43012/files?diff=unified&w=0#diff-4befd0a047da31184c43bc6fef846467a6e774ecef3cf332ac25310b3d9d4a8cL6-R9))

### CODE REVIEW

File: components/radio/index.zh-CN.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ demo:
 | defaultChecked | 初始是否选中 | boolean | false |
 | disabled | 禁用 Radio | boolean | false |
 | value | 根据 value 进行比较，判断是否选中 | any | - |
+| title | 原生 `title` 提示 | string | - | 5.7.0 |  |
```
Comment (afc163) [inline components/radio/index.zh-CN.md:48]: 不写文档了，原生属性很多，写不完。

File: components/checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -134,6 +136,7 @@ const InternalCheckbox: React.ForwardRefRenderFunction<CheckboxRef, CheckboxProp
     <label
       className={classString}
       style={style}
+      title={title}
```
Comment (afc163) [inline components/checkbox/Checkbox.tsx:139]: 下面 checkboxProps 里透传过 title，是否这个已经能生效了？

File: components/radio/group.tsx [hunk 3]
```diff
@@ -77,6 +77,7 @@ const RadioGroup = React.forwardRef<HTMLDivElement, RadioGroupProps>((props, ref
           value={option.value}
           checked={value === option.value}
           style={option.style}
+          title={option.title}
```
Comment (afc163) [inline components/radio/group.tsx:80]: 上面有个 if

File: components/radio/radio.tsx [hunk 4]
```diff
@@ -78,6 +78,7 @@ const InternalRadio: React.ForwardRefRenderFunction<CheckboxRef, RadioProps> = (
       style={style}
       onMouseEnter={props.onMouseEnter}
       onMouseLeave={props.onMouseLeave}
+      title={radioProps.title}
     >
       <RcCheckbox {...radioProps} type="radio" prefixCls={prefixCls} ref={mergedRef} />
```
Comment (afc163) [inline components/radio/radio.tsx:83]: 此处之前透传过 radioProps.title 的，是否已经有效了？

**Comment by afc163** [comment]
如果 title 已经传给过 https://github.com/react-component/checkbox/ ，是否应该修改 rc-checkbox 看看 title 为啥没挂上？

File: components/radio/group.tsx (hunk 3 above)
Comment (linxianxi) [reply components/radio/group.tsx:80]: 上面 if 是这样的 <Radio.Group options={["1", "2", "3"]} />，传不了 title，所以不用写

File: components/radio/radio.tsx (hunk 4 above)
Comment (linxianxi) [reply components/radio/radio.tsx:83]: 是生效了，但是只有按钮有，移到文字上没有 title。这个需求是否只需要移到按钮上有呢？

**Comment by linxianxi** [comment]
> 如果 title 已经传给过 https://github.com/react-component/checkbox/ ，是否应该修改 rc-checkbox 看看 title 为啥没挂上？

同上 https://github.com/ant-design/ant-design/pull/43012#discussion_r1229399238

File: components/radio/radio.tsx (hunk 4 above)
Comment (afc163) [reply components/radio/radio.tsx:83]: 确保挂到 ant-radio-button ant-radio-button-checked 这一级上就行了，现在传到里面隐藏的 input 上了。

图片

Comment (linxianxi) [reply components/radio/radio.tsx:83]: 那就是移到文字上不需要 title，只需要移到按钮上有 title。那现在这样已经是可以的，只是缺少了 ts 定义，这个 input 不是隐藏的，是 absolute 的，有位置，加了 title 是有效的，补一下 ts 应该就行了

Comment (linxianxi) [reply components/radio/radio.tsx:83]: 看错了，Radio.Button 是不行，其它的都可以（radio, checkbox）

Comment (linxianxi) [reply components/radio/radio.tsx:83]: 到时候 rc-checkbox 那边发版了，antd 这里基于 master 还是 feature？

Comment (afc163) [reply components/radio/radio.tsx:83]: 这个算 bugfix，master 就行。

Comment (linxianxi) [reply components/radio/radio.tsx:83]: rc 提 pr 了

**Comment by afc163** [comment]
https://github.com/react-component/checkbox/releases/tag/v3.1.0

**Review by afc163 — APPROVED** [review]

**Pull request merged**